### PR TITLE
Refresh Token implementation

### PR DESCRIPTION
**Pull request by VibhorCodecianGupta**
How to retrieve refresh token: 

```
GET
http://localhost:3838/oauth/authorize?response_type=token&client_id=<client ID here>&redirect_uri=http://hackerblocks.com/callback

```
The response of the above request gives a token from the url. This is the refresh token, it would last for your entire session. Retrieve it, and make the following request from the backend:

```
POST
http://localhost:3838/oauth/token

{
  "client_id" : <client ID here>,
  "redirect_uri" : "http://hackerblocks.com/callback",     
  "client_secret":"<client secret here>",
  "grant_type" : "refresh_token",
  "refresh_token":"ihb7i371gXrz9IQq5NtO7GTr7LkFk4rWPwDMGmNXtNg0k3pqOWSjwsGIbfZxhmrD"
}
```
where the value to key `refresh_token` is the token you just retrieved in the last step.

The response to this is of the form: 

```
{
    "access_token": "ahBRl6TEG5eBqeHsEEtvp3fWKDyFoxPvPoWAKalXn6lZcNek5rMINJYeFzoi7GuM",
    "refresh_token": "ihb7i371gXrz9IQq5NtO7GTr7LkFk4rWPwDMGmNXtNg0k3pqOWSjwsGIbfZxhmrD",
    "token_type": "Bearer"
}
```

Hence, retrieve the access token from here. The lifetime of this token is 10 minutes, and therefore the refresh token is to be used to send the `POST` request from the backend again (with the above-mentioned body) to obtain a fresh access token once the previous one expires.

### CODE REVIEW

**Comment by championswimmer** [comment]
Looks good. I'll test it thoroughly tomorrow!

**Comment by championswimmer** [comment]
@VibhorCodecianGupta please add a .sql file in migrations folder for migrating the token table to add the expired column :)

**Comment by championswimmer** [comment]
Also would prefer a little longer expiry times, like 24 hours.

**Comment by VibhorCodecianGupta** [comment]
Done! @championswimmer

**Comment by VibhorCodecianGupta** [comment]
@championswimmer any updates?

File: config.js [hunk 1]
```diff
@@ -17,15 +18,15 @@ config = {
   "VERIFY_EMAIL" :"3e470c46-5d2b-471e-992e-4820f1599de3"
 };
 
-config.DEPLOY_CONFIG = process.env.ONEAUTH_DEV || 'production';
+config.DEPLOY_CONFIG = process.env.ONEAUTH_DEV || 'localhost';
```
Comment (championswimmer) [inline config.js]: please don't touch configuration in a feature PR

File: config.js [hunk 2]
```diff
@@ -17,15 +18,15 @@ config = {
...
 
 switch (config.DEPLOY_CONFIG) {
 
 
   case 'localhost':
     config.SERVER_URL = 'http://localhost:3838'
     config.DEBUG = true
-    config.SECRETS = require('./secrets-sample.json')
+    config.SECRETS = require('./secrets.json')
```
Comment (championswimmer) [inline config.js]: again, same as above

File: config.js (hunk 1 above)
Comment (VibhorCodecianGupta) [reply config.js]: Ah, my bad. I was trying something out. Will revert now

**Comment by VibhorCodecianGupta** [comment]
Done! @championswimmer

**Comment by championswimmer** [comment]
i am accepting the bounty on this, the PR, i'll merge this week. need to test a few more things

**Comment by VibhorCodecianGupta** [comment]
@championswimmer does this require refactoring?

**Comment by championswimmer** [comment]
yes I intend to merge this

**Comment by VibhorCodecianGupta** [comment]
@championswimmer refactored, please have a look

**Pull request closed without merging**